### PR TITLE
SONARJAVA-3856 Handle parenthesis and assignments

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/test/files/checks/StringConcatenationInLoopCheck.java [hunk 1]
```diff
@@ -58,4 +58,17 @@ static MyObject newInstance() {
...
+      while (true) {
+        (this).f += "sdasd"; // Noncompliant
+
+        (m = this).f += "asdasd"; // not supported... what to do with this?
+        ((M) m).f += "asdas"; // not supported... what to do with this?
+      }
+    }
+  }
+
 }
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/test/files/checks/StringConcatenationInLoopCheck.java:74]: Should we move this file to the non-compiling sources folder of `java-checks-test-sources` ?

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
Simple and effective, LGTM. But we should probably move the test source file to `java-checks-test-sources`.

File: java-checks/src/test/files/checks/StringConcatenationInLoopCheck.java (hunk 1 above)
Comment (m-g-sonar) [reply java-checks/src/test/files/checks/StringConcatenationInLoopCheck.java:74]: Indeed, we should. no reason to not make it clean.

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
Great, thanks for the changes :)

**Pull request merged**